Properly compare versions with and w/o leading R letter

lemenkov opened the pull request:
Compare OTP versions with and w/o leading "R" letter properly.

kesslerm commented on src/cuttlefish.erl:
```diff
@@ -52,6 +52,10 @@ otp(DesiredMinimumOTPVersion, IfGreaterOrEqual, IfLessThan) ->
+otp([$R|TMin], TVer) ->
+    otp(TMin, TVer);
+otp(TMin, [$R|TVer]) ->
+    otp(TMin, TVer);
```
Minor nit-pick: The 'T' in 'TMin' and 'TVer' stands for the tail. So these could be renamed to 'Min' and 'Ver', if they are just the actual values.

lemenkov replied:
Done.

kesslerm commented on src/cuttlefish.erl:
```diff
@@ -135,6 +139,8 @@ otp_test() ->
     ?assert(not(otp("R16B01", "R16A"))),
     ?assert(otp("R16A", "R16A")),
+    ?assert(otp("R16A", "17.4")),
+    ?assert(not(otp("18.2.4", "17.4"))),
```
Additional test cases should include the more basic versions, too. E.g. `otp("R16", "17")`, `not(otp("17", "R16"))`, etc.

lemenkov replied:
Done. Added more tests. Also removed dotted versions, since erlang:system_info(otp_release) will never return such versions (according to man erlang).

kesslerm commented on test/erlang_vm_schema_tests.erl:
```diff
@@ -28,8 +28,8 @@ basic_schema_test() ->
     cuttlefish_unit:assert_not_configured(Config, "kernel.inet_dist_listen_max"),
-    case erlang:system_info(otp_release) of
-        [$R, $1, N|_] when N >= $6 ->
+    case cuttlefish:otp("R16", erlang:system_info(otp_release)) of
```
Using the `cuttlefish:otp/2` function here is OK, provided the unit test for this function has asserted that the function is working correctly. As an alternative, the conditional code blocks could be swapped around and guarded by `[$R, $1, N|_] when N =< $5`

lemenkov replied:
Kept as is since I've added tests for that function.

kesslerm commented on test/erlang_vm_schema_tests.erl:
```diff
@@ -88,8 +88,8 @@ override_schema_test() ->
     %% in two tests below
-    case erlang:system_info(otp_release) of
-        [$R, $1, N|_] when N >= $6 ->
+    case cuttlefish:otp("R16", erlang:system_info(otp_release)) of
```
Using the `cuttlefish:otp/2` function here is OK, provided the unit test for this function has asserted that the function is working correctly. As an alternative, the conditional code blocks could be swapped around and guarded by `[$R, $1, N|_] when N =< $5`

lemenkov replied:
Kept as is since I've added tests for that function.

lemenkov commented:
Thanks, @kesslerm. I've addressed all your notes except the last one. I suppose I've added enough tests now to consider cuttlefish:otp/2 as tested properly.

kesslerm commented:
@lemenkov Thanks. I will follow this up with the core developers.

nickelization commented:
Looks good to me, and tests pass locally. Thanks for the contribution @lemenkov!

nickelization commented:
+1 5f6c31f
